### PR TITLE
feat: URL Trimming for Twitter URL

### LINKED ISSUE #3106: Add twitter URL Trimming , Which is going 422 in CFS
Currently, when you enter a `Twitter URL` of your profile in CFS, The system expects pattern with `Complete HTTPS Protocol` which makes it throw `422` , Enable trimming in `Twitter URL` so that it's comfortable to use.

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #3106 

#### Short description of what this resolves:

- Added URL Trimming for twitter URL, As before speaker was needed to enter full Profile URL with https protocal

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by kushthedude** [comment]
@shreyanshdwivedi @Anupam-dagar @uds5501 Review Please

File: app/components/forms/session-speaker-form.js [hunk 1]
```diff
@@ -424,6 +424,11 @@ export default Component.extend(FormMixin, {
               type   : 'regExp',
               value  : validTwitterProfileUrlPattern,
               prompt : this.l10n.t('Please enter a valid twitter profile url')
+            },
+            {
+              type   : 'regExp',
+              value  : protocolLessValidUrlPattern,
+              prompt : this.l10n.t('Please enter a valid url')
```
Comment (Anupam-dagar) [inline app/components/forms/session-speaker-form.js]: I am quite not sure if we can put two same kind of rules for validation. `regExp` in this case. I am AFK as of now and hence can't test it.

File: app/templates/components/forms/session-speaker-form.hbs [hunk 2]
```diff
@@ -173,6 +173,7 @@
         <div class="field">
           <label class="{{if field.isRequired 'required'}}" for="name">{{field.name}}</label>
           {{#if (is-input-field field.type) }}
+
```
Comment (Anupam-dagar) [inline app/templates/components/forms/session-speaker-form.hbs]: why the extra newline?

File: app/components/forms/session-speaker-form.js (hunk 1 above)
Comment (kushthedude) [reply app/components/forms/session-speaker-form.js]: @Anupam-dagar I tested it for both cases, It throws an appropriate error when needed

File: app/templates/components/forms/session-speaker-form.hbs (hunk 2 above)
Comment (kushthedude) [reply app/templates/components/forms/session-speaker-form.hbs]: Typo, Will remove it

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

File: app/templates/components/forms/session-speaker-form.hbs [hunk 3]
```diff
@@ -185,7 +185,16 @@
                   canRemoveItem=false
                 }}
               {{else}}
-                {{input type=field.type value=(mut (get data.speaker field.fieldIdentifier)) id=(if field.isRequired (concat 'speaker_' field.fieldIdentifier '_required') (concat 'speaker_' field.fieldIdentifier))}}
+                {{#if (eq field.name 'Twitter')}}
```
Comment (shreyanshdwivedi) [inline app/templates/components/forms/session-speaker-form.hbs]: @kushthedude I think you can use condition `if (or (eq field.name 'Website') (eq field.name 'Twitter'))` on line 180 than duplication of code here.

File: app/templates/components/forms/session-speaker-form.hbs [hunk 4]
```diff
@@ -185,7 +185,16 @@
...
+                  {{widgets/forms/link-input
+                    fixedName=true
+                    inputId=(if field.isRequired (concat 'speaker_' field.fieldIdentifier '_required') (concat 'speaker_' field.fieldIdentifier))
+                    segmentedLink=data.speaker.segmentedLinkTwitter
```
Comment (kushthedude) [inline app/templates/components/forms/session-speaker-form.hbs]: @shreyanshdwivedi This line will be affected by the following change If i reduce the if condition according to your suggestion

Comment (shreyanshdwivedi) [reply app/templates/components/forms/session-speaker-form.hbs]: @kushthedude I think you can use an `if` condition here?

Comment (kushthedude) [reply app/templates/components/forms/session-speaker-form.hbs]: The following `if` I used before?

Comment (shreyanshdwivedi) [reply app/templates/components/forms/session-speaker-form.hbs]: Something similar to line 191. Use it with the if condition I commented earlier. This way you can merge both the code blocks into one and remove code duplication

**Comment by kushthedude** [comment]
@shreyanshdwivedi Please Review

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by mariobehling** [comment]
This was low priority and this behavior should be solved in all forms across the application. Instead of making an issue for each form field we need to solve issues in a broader way. Also, we need to focus on our milestones more. I am merging this even though priority is low. Let's not divert the attention of our devs too much on low priority issues. We need to get the system working more reliably.

**Pull request merged**